server/courselist.rb

```diff
@@ -627,6 +627,22 @@ def run_url_parse_json(new_url)
       logger.debug "#{__method__}: #{__LINE__}: #{new_url}: request_body_ruby: +++#{request_body_ruby.inspect}+++"
       request_body_ruby
     end
+
+    # Get the canvas courses from the list provided by mpathways.
```

Comment:
Having this as a separate method is more understandable and testable.  There is a lot going on in that little method.